### DISCUSSION TITLE
Should --watch work with test?

### DISCUSSION THREAD

**Discussion by alvinj** [VirtusLab/scala-cli#2043] [General]
Hi, I’m wondering if the `--watch` option should work with the `test` option? What I’m seeing is that if I save my file once, all is well, but if I change the file, it goes into a continuous/infinite loop.

Here’s my test file, which uses Munit:

```scala
//> using scala "3"
//> using dep "org.scalameta::munit::0.7.29"

def isTrue(a: Int | String | Boolean): Boolean = a match
    case 0 | "" | "0" | false => false
    case _ => true

class MyTests extends munit.FunSuite:
    test("isTrue tests") {
        assert(isTrue(1))
    }
```

And this is the command I’m running:

```sh
scala-cli test --watch Tests.scala
```

As mentioned, the first time I run that command, I see a few lines of output, and then it stops. But if I just add a blank space to the file and save it again, the command keeps running and running. The output looks like this:

```sh
> scala-cli test --watch Tests.scala
Compiling project (Scala 3.2.2, JVM)
Compiled project (Scala 3.2.2, JVM)
MyTests:
  + isTrue tests 0.019s
Watching sources, press Ctrl+C to exit, or press Enter to re-run.
Compiling project (Scala 3.2.2, JVM)
Compiled project (Scala 3.2.2, JVM)
MyTests:
  + isTrue tests 0.019s
Watching sources, press Ctrl+C to exit, or press Enter to re-run.
MyTests:
  + isTrue tests 0.018s
Watching sources, press Ctrl+C to exit, or press Enter to re-run.
MyTests:
  + isTrue tests 0.019s
Watching sources, press Ctrl+C to exit, or press Enter to re-run.
(this goes on forever ...)
```

This happens on a macOS system, regardless of the editor I use. Thanks!

**Comment by tgodzik** [comment]
Thanks for reporting, this certainly looks like a bug!

**Reply by alvinj** [reply]
You’re welcome. If you need me to make it a bug report, just let me know. I didn’t know which approach was best.

**Comment by alexarchambault** [comment]
I've been running into that bug locally, too (on macOS too).
